### PR TITLE
Fix colors option when using wildcard commands

### PR DESCRIPTION
**Pull request by daniloraisi**
This PR fix Issues #186, #210, and #234

Now when executing the command
```bash
concurrently -c "cyan,yellow" "npm:start-*"
```
will produce the same result as
```bash
concurrently -c "cyan,yellow" "npm:start-ng" "npm:start-server"
```

When mixing commands like
```bash
concurrently -k -c cyan,yellow,green "bash foo.sh" "npm:start-*"
concurrently -k -c cyan,yellow,green "npm:start-*" "bash foo.sh"
```

~~Will produce `bash foo.sh` to print as **cyan** or **yellow** respectively since the prefixColor attribution for plain commands use original index args passed to `concurrently` script.~~

Will produce `bash foo.sh` to print as **cyan** or **green**, since the attribution of prefixColor will happen after all commands are analyzed and prepared.

### CODE REVIEW

File: src/concurrently.js [hunk 1]
```diff
@@ -79,6 +86,7 @@ function mapToCommandInfo(command) {
         command: command.command || command,
         name: command.name || '',
         prefixColor: command.prefixColor || '',
```
Comment (gustavohenke) [inline src/concurrently.js:88]: we can get rid of this line now (and all its wiring)

**Review by gustavohenke — CHANGES_REQUESTED** [review]
Hey, thanks for taking this on!

> concurrently -k -c cyan,yellow,green "npm:start-*" "bash foo.sh"
> Will produce bash foo.sh to print as cyan or green,

Did you mean yellow or green? Or am I missing something in your fix..?

**Comment by daniloraisi** [comment]
> Hey, thanks for taking this on!
> 
> > concurrently -k -c cyan,yellow,green "npm:start-*" "bash foo.sh"
> > Will produce bash foo.sh to print as cyan or green,
> 
> Did you mean yellow or green? Or am I missing something in your fix..?

@gustavohenke when using this command `concurrently -k -c cyan,yellow,green "npm:start-*" "bash foo.sh"` the right thing is to color the output for `bash foo.sh` with the last color available.

Suppose you have only one script that matches `npm:start-*`. It will use cyan, so yellow will be the color used for the other command. For example, if you have two scripts that match `npm:start-*`, the last available color will be green, so `bash foo.sh` will be green.

File: src/concurrently.js (hunk 1 above)
Comment (daniloraisi) [reply src/concurrently.js]: I did that. Thanks!

**Review by gustavohenke — APPROVED** [review]
@daniloraisi I've pushed a couple of changes to your branch to make prefixes specified per command be preferred over those specified in the `prefixColors` option, plus added some docs and test.

We should be good to ship this now 🙌

**Comment by gustavohenke** [comment]
Fix out in v6.3.0.

**Pull request merged**